### PR TITLE
fix: Use a more flexible approach to check favorability of metrics

### LINKED ISSUE #1061: The check for metric name and show lower/higher is better is too strict
When reviewing another PR, I saw the following:

image

It looks like the indicator for lower/higher is better is to strict regarding the naming. I think it was introduced in this specific list:

https://github.com/probabl-ai/skore/pull/1053/files#diff-939ea0f3a7a59b88269e22268fddaecedba8d9f3212b0bc66f49e1d3bc4231caR72-R81

We should actually create a basic name without train/test/averages and have a regex that matched any metric name.

### PR DESCRIPTION
**Pull request by glemaitre**
closes #1061 

image

We are more flexible using a regular expression to check the score names. In addition, we take care to test first the `neg_` part that would mean that negative score are therefore "higher is greater" convention.

### CODE REVIEW

**Review by augustebaum — COMMENTED** [review]
The "`neg_` means higher is better" rule makes me a bit wary of false positives.
For example, `neg_accuracy` gets falsely considered as greater-is-better.

**Comment by glemaitre** [comment]
It will never happen in practice. The `neg_` is only a scikit-learn convention used to negate the loss/error/deviance such that whatever metrics used in a grid-search will be *maximized*.

So if you are a user, you are never going to defined your own `neg_` metric to make a score that lower is better.

**Comment by augustebaum** [comment]
Sounds good.

**Review by augustebaum — APPROVED** [review]
Thanks!
